### ISSUE TITLE
Unclean exits of the worker leave tasks in limbo

### ISSUE THREAD

**Issue by RealOrangeOne** [RealOrangeOne/django-tasks#46]
Currently, if a worker exits in an unclean manner, the task remains marked as "PROCESSING", and thus is never picked up for re-processing nor marked as completed / failed.

**Comment by davegaeddert** [comment]
Curious to see what solution you come up with, but in a similar project mine is to have user-configurable limit (ex. 2 hrs), and then periodically find anything that has been processing for longer than that and mark it as "LOST". Excited to see some of this make its way into Django!

**Comment by RealOrangeOne** [comment]
Having a limit, after which tasks can be picked up again seems to be the standard way forward - it's what many other queue libraries do. I'm currently looking to see if there's a way to explicitly check if the task is still running, although assuming we have "at least once" semantics, that's less important.

**Comment by mgax** [comment]
On a non-SQLite database, it would be possible to lock `DBTaskResult` rows, and the lock would be automatically released if the worker died. But I'm not sure if there's a clean way to get the same effect with SQLite (where a lock would, in fact, lock the entire database, thereby breaking concurrent workers).

**Comment by andrewgy8** [comment]
> if a worker exits in an unclean manner

Curious what you mean by this? Are you anticipating other ways of shutting down the worker that are not covered by the [shutdown](https://github.com/RealOrangeOne/django-tasks/blob/54c0cbb1f41bcfdb525fb8d0b93cdfa91e34c006/django_tasks/backends/database/management/commands/db_worker.py#L48) method?

**Comment by RealOrangeOne** [comment]
Yes, there are plenty. If the process shuts down cleanly, it's handled. If the server suddenly drops off (eg power or network issue), or a less-clean exit (eg segfault).

**Comment by jonatron** [comment]
django-background-tasks has a max run time and checks `locked_at`: [django-background-tasks models.py](https://github.com/django-background-tasks/django-background-tasks/blob/36c5c2b928d53a66ec6b9a190d16528bb4904171/background_task/models.py#L68)
django-tasks could do the same with `started_at` in `.ready`, or there could be a management command to change the status of tasks where `started_at` exceeds the max run time. If performance is important  ( #15 ) , the management command would be better, because if workers are running a query to find jobs frequently, then that query should be fast.
